### DISCUSSION TITLE
Translations / localization

### DISCUSSION THREAD

**Discussion by oscaralexander** [nmaier95/craft-shopify-product-fetcher#44] [Ideas]
I'm super excited for this native Shopify integration, and it's working great for me so far. Thanks so much for this!

I'm currently building a multilingual storefront (Craft multi-site), and while I have all my translations in place in Shopify through Translate & Adapt, the integration currently only returns results in the default language.

I would be _amazing_ if it could leverage the `currentSite.locale` property to return the appropriate translation.

**Comment by lukeholder** [comment]
Thanks for the request. Good idea. 

Just had a quick look but can't find the documentation on retrieving the other translations over the Admin API?

Could you link the Shopify docs on getting the translations over the API?

Thanks.

**Reply by oscaralexander** [reply]
Hi Luke, thanks. I'm not familiar with the [Shopify API for PHP](https://github.com/Shopify/shopify-api-php) package, which I assume the Craft integration uses, but the JS Buy SDK allows you to simply [provide a `language` parameter](https://shopify.github.io/js-buy-sdk/?shpxid=7f85bd36-29DC-4625-6533-C21688B06AA1#initializing-the-client) when initializing the API client. All subsequent requests will then automatically return results in that language, if available, or default to the default language. Works like a charm in the front end.

FWIW, client JS code looks like this:

```
const client = Client.buildClient({
    domain: 'store_name.myshopify.com',
    language: document.documentElement.lang,
    storefrontAccessToken: '...',
});
```

**Reply by oscaralexander** [reply]
[This page](https://shopify.dev/apps/internationalization) in the docs might offer some guidance too. Glancing over it, it appears the Admin API can't return translations, but the Storefront API can. Not sure which API Craft uses, but for product listings I reckon it would make sense to use the Storefront API anyway.

https://github.com/Shopify/shopify-api-php/blob/main/docs/usage/storefront.md

**Reply by vitalijalbu** [reply]
Hi guys, so basically the plugin doesn't allow us to have the multi-language shopify data, right?

**Reply by nfourtythree** [reply]
Hi All

It would be interesting to get an idea of what translated data you would like to be retrieving from Shopify.

Obviously, Craft has real strong translation and localisation features especially with multi-site. We are keen to make this plugin as fully featured as possible but at the same time not to bring too much across where potentially Craft might be better suited to solving the problem.

We are working on a new version of the plugin which will use Shopify's GraphQL admin API (which is going to be a requirement from them later this year), this might open up the door for some translation possibilities when looking at the docs: https://shopify.dev/docs/api/admin-graphql/2025-01/queries/translatableResourcesByIds

**Reply by vitalijalbu** [reply]
> Hi All
> 
> It would be interesting to get an idea of what translated data you would like to be retrieving from Shopify.
> 
> Obviously, Craft has real strong translation and localisation features especially with multi-site. We are keen to make this plugin as fully featured as possible but at the same time not to bring too much across where potentially Craft might be better suited to solving the problem.
> 
> We are working on a new version of the plugin which will use Shopify's GraphQL admin API (which is going to be a requirement from them later this year), this might open up the door for some translation possibilities when looking at the docs: https://shopify.dev/docs/api/admin-graphql/2025-01/queries/translatableResourcesByIds

I tried products and it works well

**Reply by plcdnl** [reply]
hi @nfourtythree, 
really sorry for the ping. Any news about this?
